### DISCUSSION TITLE
Obsidian Publish features

### DISCUSSION THREAD

**Discussion by JosefJezek** [withastro/starlight#1113] [Feature Requests]
### What version of `starlight` are you using?

0.13.0

### What is your idea?

Could you add these features from [Obsidian Publish](https://obsidian.md/publish)? 

- Hover previews - Display a popover preview of the linked page when you hover over it, like Wikipedia.
- Graph view - Visually explore the connections between the pages on your site.
- Stacked pages - Make links open in horizontal panes that you can browse like a stack of pages on a desk.
- Backlinks - Automatically list links to pages that reference the current page you are on.
- Wikilinks for internal links - https://help.obsidian.md/Linking+notes+and+files/Internal+links- 

### Why is this feature necessary?

These features help readers in exploring a knowledge base.

### Do you have examples of this feature in other projects?

- https://help.obsidian.md
- https://integralguide.com
- https://quartz.jzhao.xyz

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by stereobooster** [comment]
- Hover previews - more or less easy to implement, for example, [Link previews](https://astro-digital-garden.stereobooster.com/recipes/link-previews/)
- Wikilinks for internal links - can be implemented with some limits, if all files are in one folder you can use something like [remark-wiki-link-plus](https://github.com/datopian/remark-wiki-link-plus)
- Graph view and Backlinks would require some external tool which would "crawl" all the content and create graph from the content

**Reply by stereobooster** [reply]
**UPD** Backlinks https://astro-digital-garden.stereobooster.com/recipes/backlinks/

**Reply by delucis** [reply]
Hey, cool site!

**Reply by stereobooster** [reply]
**UPD** Wikilinks https://astro-digital-garden.stereobooster.com/recipes/wikilinks/

**Comment by yuhangch** [comment]
https://open.yuhang.ch/tech/
Similar idea to @stereobooster, but does not implement hover previews.

**Reply by delucis** [reply]
Very cool!

Would love to see some of these things published to npm as Starlight plugins if people are interested. Happy to answer questions if people have them.

**Comment by stereobooster** [comment]
Also worth to mention @HiDeoo https://github.com/HiDeoo/starlight-obsidian

**Comment by stereobooster** [comment]
I figured out how to implement [Obsidian Dataview](https://blacksmithgu.github.io/obsidian-dataview/) in Astro (generally in any SSG that uses remark). See https://astro-digital-garden.stereobooster.com/recipes/obsidian-dataview/

**Reply by HuakunShen** [reply]
I can't get it working, probably because astro has updated? Can you update this package?

**Comment by HiDeoo** [comment]
Considering many suggestions have been listed to handle this use case, either with recipes, remark plugins, or Starlight plugins mentioned in this discussion, I don't think we have plans to implement this in Starlight core. Considering this, I'll close this discussion for now.
